firmware/robot2015/src-ctrl/modules/ControllerTaskThread.cpp

```diff
@@ -0,0 +1,73 @@
+#include "controller.hpp"
+
+#include <rtos.h>
+#include <logger.hpp>
+
+#include "motors.hpp"
+#include "mpu-6050.hpp"
+
+
+// Keep this pretty high for now. Ideally, drop it down to ~3 for production builds. Hopefully that'll be possible without the console
+#define CONTROL_LOOP_WAIT_MS 250
+
+
+/**
+ * initializes the motion controller thread
+ */
+void Task_Controller(void const* args)
+{
+	// Store the thread's ID
+	osThreadId threadID = Thread::gettid();
+
+	// Store our priority so we know what to reset it to if ever needed
+	osPriority threadPriority;
+
+	if (threadID != NULL)
+		threadPriority  = osThreadGetPriority(threadID);
+	else
+		threadPriority = (osPriority)NULL;
+
+	float gyroVals[3] = { 0 };
+	float accelVals[3] = { 0 };
+
+	MPU6050 imu(RJ_I2C_BUS);
```

Comment:
It may be better if we initialize the MPU in main and pass it in here.  That way we can run the imu and get readings out of it without having to run the control loop.